cirq-google/cirq_google/api/v2/program.py

```diff
@@ -59,8 +76,26 @@ def qubit_from_proto_id(proto_id: str) -> cirq.Qid:
     Returns:
         A `cirq.Qid` corresponding to the proto id.
     """
-    num_coords = len(proto_id.split('_'))
-    if num_coords == 2:
+    # Avoid circular import
+    from cirq_google.devices.coupler import Coupler
+
+    qubit_field = proto_id.split('_')
+    num_coords = len(qubit_field)
+    if proto_id[:2] == 'c_':
+        if num_coords == 5:
+            # 2 grid qubits: c_2_1_4_3
+            grid_qubit0_str = qubit_field[1] + '_' + qubit_field[2]
+            grid_qubit1_str = qubit_field[3] + '_' + qubit_field[4]
+            grid_qubit0 = grid_qubit_from_proto_id(grid_qubit0_str)
+            grid_qubit1 = grid_qubit_from_proto_id(grid_qubit1_str)
+            return Coupler(grid_qubit0, grid_qubit1)
+        elif num_coords == 3:
+            # 2 line qubits: c_2_4
+            # Or two named qubits: c_qubita_qubitb
+            line_qubit0 = qubit_from_proto_id(qubit_field[1])
+            line_qubit1 = qubit_from_proto_id(qubit_field[2])
```

Comment:
LGTM just catch the `ValueError` in the Coupler case and `pass` so it doesn't fail and instead creates a `NamedQubit`